tests/v2/validation/nodemetadata/node_annotation_test.go

```diff
@@ -0,0 +1,286 @@
+//go:build (validation || infra.any || cluster.any || extended) && !sanity && !stress
+
+package nodemetadata
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type NodeAnnotationTestSuite struct {
+	suite.Suite
+	client    *rancher.Client
+	session   *session.Session
+	cluster   *management.Cluster
+	clusterID string
+}
+
+func (ann *NodeAnnotationTestSuite) TearDownSuite() {
+	ann.session.Cleanup()
+}
+
+func (ann *NodeAnnotationTestSuite) SetupSuite() {
+	ann.session = session.NewSession()
+
+	client, err := rancher.NewClient("", ann.session)
+	require.NoError(ann.T(), err)
+
+	ann.client = client
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(ann.T(), clusterName, "Cluster name should be set")
+
+	ann.clusterID, err = clusters.GetClusterIDByName(ann.client, clusterName)
+	require.NoError(ann.T(), err, "Error getting cluster ID")
+
+	ann.cluster, err = ann.client.Management.Cluster.ByID(ann.clusterID)
+	require.NoError(ann.T(), err)
+}
+
+func (ann *NodeAnnotationTestSuite) TestAddAnnotation() {
+
+	log.Info("Verify user is able to add annotation on the node through API")
+
+	steveclient, err := ann.client.Steve.ProxyDownstream(ann.clusterID)
+	require.NoError(ann.T(), err)
+
+	query, err := url.ParseQuery("labelSelector=node-role.kubernetes.io/worker=true")
+	require.NoError(ann.T(), err)
+
+	nodeList, err := steveclient.SteveType("node").List(query)
+	require.NoError(ann.T(), err)
+
+	nodeSpec, err := getNodeSpec(nodeList)
+	require.NoError(ann.T(), err)
+
+	key := namegen.AppendRandomString("key")
+	value := namegen.AppendRandomString("value")
+
+	err = addAnnotation(nodeSpec, nodeList, steveclient, key, value)
+	require.NoError(ann.T(), err)
+
+	result, nodeList, err := verifyAnnotation(query, steveclient, true, key, value)
+	require.NoError(ann.T(), err)
+	require.Falsef(ann.T(), result == false, "Annotation is not added on the node")
+
+	nodeSpec, err = getNodeSpec(nodeList)
+	require.NoError(ann.T(), err)
+
+	err = deleteAnnotation(nodeSpec, nodeList, steveclient, key)
+	require.NoError(ann.T(), err)
+
+	result, nodeList, err = verifyAnnotation(query, steveclient, false, key, value)
+	require.NoError(ann.T(), err)
```

Comment:
a lot of this seems reused; I  would think a helper that encapsulates this would be helpful. Since you're adding an annotation in pretty much every one of these tests. Something like an `addAndVerifyAnnotation` would make sense to me, what do you think?

Reply:
Verify is common to both Add and Delete. So i don't want to write a verify function just for Add. 

All of this is encapsulated as functions. It needs an error check, which makes it a lot more longer. But I feel the test is to the point, and needs no further changes.

Reply:
this makes the test more verbose and more difficult to maintain in the longrun. Having reusable functions would be ideal to avoid copy-pasta in tests.

Reply:
I understand and agree. But, all these are in functions already.